poh/src/poh_service.rs

```diff
@@ -454,11 +461,27 @@ mod tests {
                     loop {
                         // send some data
                         let mut time = Measure::start("record");
-                        let _ =
+                        let res =
```

Comment:
cache result so we don't deadlock trying to acquire poh_recorder write lock while holding it